homeassistant/components/upnp/config_flow.py

```diff
@@ -10,26 +11,80 @@
 from homeassistant import config_entries
 from homeassistant.components import ssdp
 from homeassistant.const import CONF_SCAN_INTERVAL
-from homeassistant.core import callback
+from homeassistant.core import HomeAssistant, callback
 
 from .const import (
     CONFIG_ENTRY_HOSTNAME,
     CONFIG_ENTRY_SCAN_INTERVAL,
     CONFIG_ENTRY_ST,
     CONFIG_ENTRY_UDN,
     DEFAULT_SCAN_INTERVAL,
-    DISCOVERY_HOSTNAME,
-    DISCOVERY_LOCATION,
-    DISCOVERY_NAME,
-    DISCOVERY_ST,
-    DISCOVERY_UDN,
-    DISCOVERY_UNIQUE_ID,
-    DISCOVERY_USN,
     DOMAIN,
     DOMAIN_DEVICES,
     LOGGER as _LOGGER,
+    SSDP_SEARCH_TIMEOUT,
+    ST_IGD_V1,
+    ST_IGD_V2,
 )
-from .device import Device, discovery_info_to_discovery
+
+
+def _friendly_name_from_discovery(discovery_info: Mapping[str, Any]) -> str:
+    """Extract user-friendly name from discovery."""
+    if discovery_info.get("friendlyName"):
+        return discovery_info["friendlyName"]
+
+    if discovery_info.get("modeName"):
+        return discovery_info["modelName"]
+
+    return discovery_info.get("_host", "")
```

Comment:
Personal preference, but this could be written as:

```suggestion
    return discovery_info.get("friendlyName") or discovery_info.get("modeName") or discovery_info.get("_host", "")
```